src/include/homestore/meta_service.hpp

```diff
@@ -87,6 +88,7 @@ class MetaBlkService {
     MetablkMetrics m_metrics;
     bool m_inited{false};
     std::unique_ptr< meta_vdev_context > m_meta_vdev_context;
+    std::unordered_map< meta_sub_type, std::vector< meta_sub_type > > m_dependency_topological_graph;
```

Comment:
Can we move this to m_sub_info which mantains all the registration info for all clients?

Reply:
step 2 is not that easy, B may have a dependency C (assuming it already registered ), and D may depends on B((assuming it also registered).  Now you need to insert B after C, but before A/D.

Actually you are doing topo-sort, in another form, standard topo-sort algorithm is much easier to follow.

Reply:
@yamingk we do not know what kind of dependency(single or multi) will we meet in the future,  i think topological sort is thorough solution

Reply:
I think the logic will be much more easier to follow if it is one-to-one dependency. But it could also be subjective. 
Bottom line is I am okay with any solution (algorithm) to solve the current problem (which is a one-to-one dependency), but we should not be solving problems that there is no real use case.